rotkehlchen/accounting/structures/processed_event.py

```diff
@@ -117,6 +123,21 @@ def to_exported_dict(
                 exported_dict['asset'] = ''
             if tx_hash is not None:
                 exported_dict['notes'] = f'{evm_explorer}{tx_hash}  ->  {self.notes}'
+            if self.location in EVM_LOCATIONS and database is not None:
+                def _maybe_add_label_with_address(matched_address: re.Match[str]) -> str:
+                    """Adds a label of the address if it exists in a given string"""
+                    chain_address = ChainAddress(
+                        address=string_to_evm_address(matched_address.group(0)),
+                        blockchain=ChainID(self.location.to_chain_id()).to_blockchain(),
+                    )
+                    name = database.get_blockchain_account_label(chain_address)
+                    return f'{chain_address.address} [{name}]' if name else chain_address.address
+                exported_dict['notes'] = re.sub(
+                    # call _maybe_add_label_with_address on each address in the note
+                    pattern=r'\b0x[a-fA-F0-9]{40}\b',
+                    repl=_maybe_add_label_with_address,
+                    string=str(exported_dict['notes']),
```

Comment:
isn't this already a string? why the cast to string?

Reply:
mypy doesn't like it because, `exported_dict` was of type `dict[str, Any]`

Reply:
we never make more work on the production code to satisfy typing. You should fix typing instead str() on a string is wasting resources.